modules/i18n/input-method/default.nix

```
@@ -0,0 +1,68 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.i18n.inputMethod;
+
+  gtk2_cache = pkgs.runCommand "gtk2-immodule.cache" {
+    preferLocalBuild = true;
+    allowSubstitutes = false;
+    buildInputs = [ pkgs.gtk2 cfg.package ];
+  } ''
+    mkdir -p $out/etc/gtk-2.0/
+    GTK_PATH=${cfg.package}/lib/gtk-2.0/ gtk-query-immodules-2.0 > $out/etc/gtk-2.0/immodules.cache
+  '';
+
+  gtk3_cache = pkgs.runCommand "gtk3-immodule.cache" {
+    preferLocalBuild = true;
+    allowSubstitutes = false;
+    buildInputs = [ pkgs.gtk3 cfg.package ];
+  } ''
+    mkdir -p $out/etc/gtk-3.0/
+    GTK_PATH=${cfg.package}/lib/gtk-3.0/ gtk-query-immodules-3.0 > $out/etc/gtk-3.0/immodules.cache
+  '';
+
+in {
+  options.i18n = {
+    inputMethod = {
+      enabled = mkOption {
+        type = types.nullOr
+          (types.enum [ "fcitx" "fcitx5" "nabi" "uim" "hime" "kime" ]);
+        default = null;
+        example = "fcitx";
+        description = ''
+          Select the enabled input method. Input methods is a software to input symbols that are not available on standard input devices.
+
+          Input methods are specially used to input Chinese, Japanese and Korean characters.
+
+          Currently the following input methods are available in NixOS:
```

```suggestion
          Currently the following input methods are available in Home Manager:
```
(Unfortunately) we need to update the "branding" :smile: